great_expectations/checkpoint/checkpoint.py

```diff
@@ -468,82 +468,6 @@ def _run_validation(  # noqa: PLR0913
                 f"Exception occurred while running validation[{idx}] of Checkpoint '{self.name}': {e.message}."
             ) from e
 
-    def self_check(self, pretty_print: bool = True) -> dict:
```

Comment:
This method is only ever used in `test_yaml_config` as a way to valid an object's config. This can be removed.